test/controllers/util/CustomMetadataUtilsSpec.scala

```diff
@@ -136,13 +138,26 @@ class CustomMetadataUtilsSpec extends AnyFlatSpec with MockitoSugar with BeforeA
             verifyBoolean(field.asInstanceOf[RadioButtonGroupField], property.defaultValue)
           case Text => field.isInstanceOf[DropdownField] should be(true)
             verifyText(field.asInstanceOf[DropdownField], property)
+          case unknownType => throw new IllegalArgumentException(s"Invalid type $unknownType")
         }
         field.fieldDescription should equal(property.description.getOrElse(""))
         field.fieldName should equal(property.fullName.get)
         field.isRequired should equal(property.propertyGroup.contains("MandatoryMetadata"))
     }
   }
 
+  "convertPropertiesToFields" should "convert date property to field and it shouldn't allow future date when property name is foiExemptionAsserted" in {
```

Comment:
Similar comment to the one I made at the very top about hardcoding foiExemptionAsserted

Reply:
Yeah, we will change this one if we go with a data-driven approach.